File: neoforge/src/main/java/net/shiroha233/roadweaver/client/neoforge/ClientEvents.java
package net.shiroha233.roadweaver.client.neoforge;

import net.neoforged.neoforge.client.event.ClientPlayerNetworkEvent;
import net.neoforged.neoforge.common.NeoForge;
import net.shiroha233.roadweaver.client.map.data.MapSnapshotCache;

public final class ClientEvents {
    private ClientEvents() {}

    public static void register() {
        NeoForge.EVENT_BUS.addListener(ClientEvents::onLoggingIn);
        NeoForge.EVENT_BUS.addListener(ClientEvents::onLoggingOut);
    }

    private static void onLoggingIn(ClientPlayerNetworkEvent.LoggingIn e) {
        MapSnapshotCache.clearNow();
    }

    private static void onLoggingOut(ClientPlayerNetworkEvent.LoggingOut e) {
        MapSnapshotCache.clearNow();
    }
}

File: neoforge/src/main/kotlin/net/shiroha233/roadweaver/client/neoforge/ClientEvents.kt
package net.shiroha233.roadweaver.client.neoforge

import net.neoforged.neoforge.client.event.ClientPlayerNetworkEvent
import net.neoforged.neoforge.common.NeoForge
import net.shiroha233.roadweaver.client.map.data.MapSnapshotCache

@Suppress("MemberVisibilityCanBePrivate")
object ClientEvents {
    @JvmStatic
    fun register() {
        NeoForge.EVENT_BUS.addListener(::onLoggingIn)
        NeoForge.EVENT_BUS.addListener(::onLoggingOut)
    }

    private fun onLoggingIn(e: ClientPlayerNetworkEvent.LoggingIn) {
        MapSnapshotCache.clearNow()
    }

    private fun onLoggingOut(e: ClientPlayerNetworkEvent.LoggingOut) {
        MapSnapshotCache.clearNow()
    }
}
